Mathlib/RingTheory/SimpleModule.lean

```diff
@@ -90,8 +101,55 @@ theorem isAtom : IsAtom m :=
   isSimpleModule_iff_isAtom.1 hm
 #align is_simple_module.is_atom IsSimpleModule.isAtom
 
+variable [IsSimpleModule R M]
+open LinearMap
+
+theorem span_singleton_eq_top {m : M} (hm : m ≠ 0) : Submodule.span R {m} = ⊤ :=
+  (eq_bot_or_eq_top _).resolve_left fun h ↦ hm (h.le <| Submodule.mem_span_singleton_self m)
+
+-- needs import to use Submodule.IsPrincipal
```

Comment:
I feel that definition of Submodule.IsPrincipal should be moved to another (more basic) file so that this import is not necessary.

Reply:
I agree! I think I missed the opportunity in #10327, but both that PR and this one are already large. Hopefully there will be another opportunity soon ...

Reply:
Well, that's something that will be forgotten in two days. Maybe a simple invocation of #find_home! will indicate you the place to move this.

Reply:
Probably I should also move IsPrincipalIdealRing earlier (or create a new file PrincipalIdealDomain/Defs) so I can also add the IsSemisimpleRing -> IsPrincipalIdealRing instance ...

Reply:
done now